src/journal/journal-file.c

```diff
@@ -269,7 +269,7 @@ static int journal_file_set_online(JournalFile *f) {
         if (!f->writable)
                 return -EPERM;
 
-        if (!(f->fd >= 0 && f->header))
```

Comment:
This same exact check is done in `journal_file_set_offline()`, if you're going to simplify it at least do it for the whole listing for consistency sake...